simulations/src/streaming/mod.rs

```diff
@@ -253,7 +253,7 @@ where
         })
     }
 
-    pub fn stop(self) -> anyhow::Result<()> {
+    pub fn stop(&self) -> anyhow::Result<()> {
```

Comment:
Why are we using a `&` here now? If we need to somehow wait till they finish, this is a good place to do it.

Reply:
It was added because in [this diff](https://github.com/logos-co/nomos-node/pull/305/files#diff-a0c70343d4d8e44d15d7b662fbeac879ee57546748aed66d60ec92e1b01d8fb0R181-R193) handle would be moved in either branch otherwise.
Another solution is to make both `stop` and `shutdown` methods self consuming and then wrap it into `Arc<Option<Handler>>` in the `signal` handler and `take` it when simulation has to stop.

Reply:
Its ok if it is easier this way. Just wanted to understand it. Thanks!

Reply:
> If we need to somehow wait till they finish, this is a good place to do it.

These calls are blocking, simulation exits when `stop` or `shutdown` functions finishes, I think this part is working ok.